File: VB/Q200298/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB
Imports DXSample
Imports DevExpress.XtraEditors.Controls

Namespace Q200298

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            XpoDefault.ConnectionString = AccessConnectionProvider.GetConnectionString("..\..\nwind.mdb")
            InitializeComponent()
        End Sub

        Private Sub OnApplyFilterClick(ByVal sender As Object, ByVal e As EventArgs)
            filterControl1.ApplyFilter()
        End Sub

        Private Sub OnhProductsCustomDisplayText(ByVal sender As Object, ByVal e As CustomDisplayTextEventArgs)
            If Equals(e.DisplayText, String.Empty) Then
                Dim o As Object = e.Value
            End If
        End Sub

        Private Sub filteredXPComponent1_CustomFilterColumnEditor(ByVal sender As Object, ByVal e As CustomHelperFilterColumnEditorEventArgs)
            If Equals(e.Column.FieldName, "ProductID.CategoryID!Key") Then e.Editor = leCategories
        End Sub
    End Class
End Namespace

File: CS/Q200298/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using DevExpress.XtraEditors.Filtering;
using DXSample;
using Northwind;
using DevExpress.Xpo.Metadata;
using DevExpress.XtraEditors.Controls;

namespace Q200298 {
    public partial class Form1 : Form {
        public Form1() {
            XpoDefault.ConnectionString = AccessConnectionProvider.GetConnectionString(@"..\..\nwind.mdb");
            InitializeComponent();
        }

        private void OnApplyFilterClick(object sender, EventArgs e) {
            filterControl1.ApplyFilter();
        }

        private void OnhProductsCustomDisplayText(object sender, CustomDisplayTextEventArgs e) {
            if (e.DisplayText == string.Empty) {
                object o = e.Value;
            }
        }

        private void filteredXPComponent1_CustomFilterColumnEditor(object sender, CustomHelperFilterColumnEditorEventArgs e) {
            if (e.Column.FieldName == "ProductID.CategoryID!Key")
                e.Editor = leCategories;
        }
    }
}